compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1515,13 +1521,28 @@ public void visit(BLangTypeDefinition typeDefinition) {
         }
 
         typeDefinition.setPrecedence(this.typePrecedence++);
-        BTypeSymbol typeDefSymbol;
-
-        boolean label = false;
-        if (definedType.tsymbol.name != Names.EMPTY) {
-            typeDefSymbol = definedType.tsymbol.createLabelSymbol();
-            label = true;
-        } else {
+        BSymbol typeDefSymbol;
```

Comment:
I think we should be able to further simplify this visit method 🤔

Reply:
Working on cleaning the old workarounds here. In the meantime, please review the rest

Reply:
Will handle cleaning the old workarounds in https://github.com/ballerina-platform/ballerina-lang/issues/32848